docs/pallas/gpu/pipelining.md

```diff
@@ -0,0 +1,328 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    main_language: python
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.16.4
+  kernelspec:
+    display_name: Python 3
+    name: python3
+---
+
+<!-- #region id="bJ5yuIr-M0x0" -->
+## Mosaic GPU Pipelining
+
+This guide covers software pipelining using the Mosaic GPU backend for Pallas.
+
+For a general overview of the pipelining API in Pallas, we recommend that users first read the [Pallas Pipelining API](https://docs.jax.dev/en/latest/pallas/pipelining.html#pallas-pipelining-api). Pipelining in Pallas is programmed explicitly. For those who are familiar with Triton, this is a significant difference in programming model because in Triton, pipelining is an optimization that is inferred automatically by the compiler.
+
+<!-- #endregion -->
+
+```python id="dGAa3iO5DoRT"
+import jax
+from jax import lax
+from jax import numpy as jnp
+from jax.experimental.pallas import mosaic_gpu as plgpu
+from jax.experimental import pallas as pl
+import numpy as np
+```
+
+<!-- #region id="Pv9j90hVyswo" -->
+
+### Pipelining with Mosaic GPU
+
+
+While the Mosaic GPU backend supports pipelining via `pl.pallas_call`, the recommended approach is to use the `emit_pipeline` function to pipeline over sequential loops, and to use `pl.pallas_call` to partition the problem in parallel over the CUDA grid. `emit_pipeline` follows the same API as `pl.pallas_call` except it exposes a few additional GPU-specific options.
+
+```python
+def emit_pipeline(
```

Comment:
We should probably link to the API docs instead of duplicating the signature here? I have a mild preference for moving parameter docs into the docstrings as well.

Reply:
Done.